features/hooks.feature

```diff
@@ -63,6 +63,16 @@ Feature: Environment Hooks
     Then it fails
     And the "After" hook has status "passed"
 
+  Scenario: Failing before step fails the scenario
+    Given a file named "features/support/hooks.js" with:
+      """
+      import {BeforeStep} from 'cucumber'
+
+      BeforeStep(function() { throw 'Fail' })
+      """
+    When I run cucumber-js
+    Then it fails
```

Comment:
Can we please put these new features in their own feature file? I think we will have enough new features by the end to warrant their own file. The world in hooks would get its own step hook feature